test/distributed/test_c10d.py

```diff
@@ -2989,6 +2997,139 @@ def test_param_layout_mismatch_error(self):
             with self.assertRaisesRegex(RuntimeError, ".* appears not to match strides of the same param in process 0"):
                 m_ddp = DistributedDataParallel(m, device_ids=[dev0], process_group=process_group)
 
+    @requires_gloo()
+    def test_ddp_comm_hook_future_passing_cpu(self):
+        """
+        This unit test verifies whether the Future object is passed properly.
+        The callback function creates a Future object and sets a value to it.
+        """
+        store = c10d.FileStore(self.file_name, self.world_size)
+        process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size)
+
+        # Test on CPU
+        cpu_model = DistributedDataParallel(
+            TestDdpCommHook().cpu(),
+            process_group=process_group
+        )
+
+        # Register DDP Communication Hook
+        cpu_model._register_comm_hook(None, self._simple_hook)
+
+        self._run_and_verify_simple_hook(cpu_model)
+
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_ddp_comm_hook_future_passing_gpu(self):
```

Comment:
This test should be able to run with both nccl and gloo since gloo supports the GPU.